traits/tests/test_deprecated_handlers.py

```diff
@@ -24,14 +27,21 @@ def test_handler_warning(self):
         handlers = {
             "TraitDict": TraitDict,
             "TraitList": TraitList,
-            "TraitTuple": TraitTuple
+            "TraitTuple": TraitTuple,
+            "TraitMap": [TraitMap, {}],
```

Comment:
You could avoid having to alter the code below by keeping the `handler_factory` idea and using `lambda: TraitMap({})` here (and similarly for the next two lines).  Alternatively, change the existing entries (`TraitDict`, ...) to use the same format.  Using two different formats in this list and then using an `isinstance` check to distinguish them is something of an antipattern.

Reply:
Changed to use the first approach suggested. Thanks!